### ISSUE TITLE
Dependency ordering issue with AWS security groups defined in nested modules.

### ISSUE THREAD

**Issue by kyriakosoikonomakos** [hashicorp/terraform#669]
Steps to reproduce:
 * Define a "common" module that creates a security group.
 * Define an "app" module that imports the "common" module and defines another security group.
 * Set the SG created in common as source for the ingress rule in the "app" SG.
 * Create a top level .tf file and call the app module.
 * "Apply" the environment
 * "Destroy" the environment.

Current behavior: 
You will notice that terraform will first attempt to destroy the SG defined in common without destroying the "app" SG first. Terraform will just hang there for some time and eventually exit with an error.

Expected behavior: 
Terraform should first remove "app" SG and then "common" SG.

Sample output:
```
➜  /tmp/tmp_tf  ./bin/terraform destroy -var-file=dev.tfvars -force=true
module.app.common.aws_security_group.td_common_sg: Refreshing state... (ID: sg-988115fd)
module.app.aws_security_group.app_sg: Refreshing state... (ID: sg-9b8115fe)
module.app.common.aws_security_group.td_common_sg: Destroying...
module.app.common.aws_security_group.td_common_sg: Error: resource sg-988115fd has a dependent object (DependencyViolation)
Error applying plan:

1 error(s) occurred:

* resource sg-988115fd has a dependent object (DependencyViolation)

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
➜  /tmp/tmp_tf
```
Workaround:
Create the common SG in the top level and then pass it as an input to the modules.

Sample code and state files can be found in https://github.com/kyriakosoikonomakos/terraform-issue-example

**Comment by salizzar** [comment]
Hi @kyriakosoikonomakos,

Do you have a example of these workaround, based on your terraform-issue-example? I have the exact problem here.

Thank you.

**Comment by kyriakosoikonomakos** [comment]
@salizzar It looks like this has been fixed at 0.3.6. At least my test repository can now be created and destroyed just fine. 

@armon is it OK for me to close this bug report?

**Comment by armon** [comment]
Yep I can do it!
